### PR TITLE
cleanup table with clean history

### LINKED ISSUE #1128: Cleanup Table
from this PRs comments: https://github.com/axa-ch/patterns-library/pull/1127

TODOs:
- Add better test data (texts that make sense)
- Fix @raphaellueckl comment: "I would extract this part into a named method, since I have no idea what this is doing."

### PR DESCRIPTION
**Pull request by danyball**
fixes #1128

### CODE REVIEW

File: src/components/30-organisms/table-sortable/unit.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import AXATableSortable from './index';
+
+describe('validateModel()', () => {
+  test('should return true if no elements are given', () => {
```
Comment (LucaMele) [inline src/components/30-organisms/table-sortable/unit.test.js]: Please add a test where we check if footer is optional, something like so:

```
AXATableSortable.prototype.model = {
      thead: [],
      tbody: [[]],
    };

     expect(AXATableSortable.prototype.validateModel()).toBe(true);
```

**Comment by danyball** [comment]
These tests failed with one-liner-if, but i they have to be green?

```
 ✕ should return true if tbody has no elements, but thead and tfoot each have one (26ms)
    ✕ should return true if thead has no elements, but tfoot and tbody each have one (1ms)
```

**Comment by danyball** [comment]
@LucaMele i refactored the validation logic (see last comment) and fixed a bug in line 97 `tfoot[0].length`. The PR is ready to review again.

**Comment by LucaMele** [comment]
Hello @danyball , tbody is mandatory and tfoot makes no sense to exist without tbody, therefore the test should be: `should return false if tbody has no elements, but thead and tfoot each have one`

**Comment by LucaMele** [comment]
So to answer:

```
 ✕ should return true if tbody has no elements, but thead and tfoot each have one (26ms) -> FALSE
 ✕ should return true if thead has no elements, but tfoot and tbody each have one (1ms) -> I guess TRUE
 ✕ should return true if tbody is set but nothing else (6ms) -> I guess TRUE
```

**Pull request merged**